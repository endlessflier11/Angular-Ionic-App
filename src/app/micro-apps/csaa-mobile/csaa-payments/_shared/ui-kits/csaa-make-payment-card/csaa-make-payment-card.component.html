<csaa-skeleton-card *ngIf="loading"></csaa-skeleton-card>
<ion-card
  *ngIf="!loading && payment && !paymentSucceeded"
  [class.overdue]="isOverdue()"
  class="csaa-card"
>
  <ion-card-header>
    <ion-card-title>{{ payment.policyNumber }}</ion-card-title>
  </ion-card-header>
  <ion-card-content>
    <ion-grid>
      <ion-row class="item">
        <ion-col>
          <div class="policy-meta-content">
            <h3>
              <ng-container *ngIf="isOverdue()">Past Due</ng-container>
              <ng-container *ngIf="!isOverdue()">Due:</ng-container>
              {{ payment.dueDate | date: 'mediumDate':timezone }}
            </h3>
            <span class="subtitle">
              <csaa-policy-subtitles [policySubtitle]="payment.subtitle"></csaa-policy-subtitles>
            </span>
          </div>
        </ion-col>
        <ion-col size="auto">
          <div class="policy-meta-icon">
            <csaa-theme-icon slot="end" [type]="payment.policyType" [size]="40"></csaa-theme-icon>
          </div>
        </ion-col>
      </ion-row>
    </ion-grid>

    <ion-list>
      <ion-item
        detail
        class="csaa-payment-amount-list-item ion-no-padding"
        (click)="selectAmountClick.emit(payment)"
      >
        <ion-label>Amount </ion-label>
        <span slot="end">{{
          payment.otherAmount || payment.amount | currency: 'USD':'symbol':'1.2-2'
        }}</span>
        <!--  payment.paid is unknown
                <span slot="end" *ngIf="!payment.paid">{{
          payment.otherAmount || payment.amount | currency: 'USD':'symbol':'1.2-2'
        }}</span>
        <span slot="end" *ngIf="payment.paid">{{ 0 | currency: 'USD':'symbol':'1.2-2' }}</span> -->
      </ion-item>
      <ion-item
        *ngIf="!payment.autoPay"
        detail
        lines="none"
        class="csaa-payment-method-list-item ion-no-padding"
        (click)="selectPaymentClick.emit(payment)"
      >
        <ion-avatar slot="start" *ngIf="paymentMethod">
          <span>
            <!-- Span first because ion-avatar::slotted(img) style forces image to render unexpectedly -->
            <img
              *ngIf="paymentMethod?.card?.type"
              [src]="'/assets/csaa-mobile/vectors/' + paymentMethod.card.type + '.svg'"
            />
            <img *ngIf="paymentMethod?.account" [src]="'/assets/csaa-mobile/vectors/bank.svg'" />
          </span>
        </ion-avatar>
        <ion-label
          [ngClass]="{ 'card-expired': paymentMethod?.card && paymentMethod?.card.expired }"
          class="ion-text-nowrap"
        >
          <span>{{ paymentMethod ? paymentMethod.shortName : 'Pay With' }}</span>
        </ion-label>
        <span slot="end" *ngIf="paymentMethod?.card && paymentMethod?.card?.isDebitCard !== true">
          {{ 'Credit' }}
        </span>
        <span slot="end" *ngIf="paymentMethod?.card?.isDebitCard">{{ 'Debit' }}</span>
        <span slot="end" *ngIf="paymentMethod?.account">{{
          paymentMethod?.account?.type | titlecase
        }}</span>
        <span slot="end" *ngIf="!paymentMethod">Payment Method</span>
      </ion-item>
      <ion-item
        detail
        class="csaa-payment-autopay-item ion-no-padding"
        *ngIf="hasRegId()"
        (click)="goToAutopaySettings()"
      >
        <ion-label>
          <label>AutoPay</label>
          <small *ngIf="payment.autoPay">
            {{ getAutopayInfo() }}
          </small>
        </ion-label>
        <span slot="end"> {{ payment.autoPay ? 'On' : 'Off' }}</span>
      </ion-item>
    </ion-list>

    <ion-button expand="full" (click)="makePaymentClick.emit(payment)" class="ion-no-margin">
      Make a Payment
    </ion-button>
  </ion-card-content>
</ion-card>
