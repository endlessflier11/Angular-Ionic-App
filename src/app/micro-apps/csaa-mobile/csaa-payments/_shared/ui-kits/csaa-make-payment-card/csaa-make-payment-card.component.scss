@import '../../../../_scss/variables';
.csaa-card {
  ion-card-content {
    ion-grid {
      margin-bottom: 15px;
      .subtitle {
        font-weight: $csaa-light-font-weight;
      }
    }

    ion-list {
      margin-bottom: 20px;
    }

    ion-item.csaa-payment-amount-list-item,
    ion-item.csaa-payment-autopay-item {
      font-size: 16px !important;
      > span {
        margin-inline-end: 5px;
        color: $csaa-lighter-color;
      }
    }

    ion-item.csaa-payment-autopay-item {
      ion-label {
        display: flex !important;
        flex-flow: column nowrap;
        small {
          margin-top: 3px;
          color: $csaa-lighter-color;
        }
      }
    }
    ion-item.csaa-payment-method-list-item {
      font-size: 16px !important;
      border-bottom: $csaa-list-item-border;

      ion-avatar {
        margin-inline-end: 5px;
        > span {
          width: 30px;
          display: flex;
          align-items: center;
          height: 100%;
        }

        img {
          max-width: 100%;
        }
      }

      ion-label {
        margin-inline-end: 5px;
        > span {
          display: block;
          width: 100%;
          text-overflow: ellipsis;
          overflow: hidden;
        }
      }

      > span {
        color: $csaa-lighter-color;
        margin-inline-start: 10px;
        margin-inline-end: 5px;
      }
    }
  }

  &.overdue {
    ion-card-title {
      color: $csaa-emergency-color;
    }
    ion-card-content {
      ion-grid {
        ion-row:first-child {
          h3 {
            color: $csaa-emergency-color;
          }
        }
      }
    }
  }
}
