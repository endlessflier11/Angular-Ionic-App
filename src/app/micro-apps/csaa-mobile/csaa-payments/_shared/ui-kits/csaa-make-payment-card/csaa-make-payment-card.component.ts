import { Component, EventEmitter, Input, OnChanges, Output, SimpleChanges } from '@angular/core';
import {
  UpcomingPayment,
  CustomerSearchResponse,
  PaymentAccount,
  PolicyType,
  AutoPayEnrollmentResponse,
  EventName,
  Category,
  EventType,
} from '../../../../_core/interfaces';
import { PaymentHelper } from '../../../../_core/shared/payment.helper';
import { AnalyticsService, RouterService } from '../../../../_core/services';
import { noop } from '../../../../_core/helpers';
import { AlertController } from '@ionic/angular';

@Component({
  selector: 'csaa-make-payment-card',
  templateUrl: 'csaa-make-payment-card.component.html',
  styleUrls: ['csaa-make-payment-card.component.scss'],
})
export class CsaaMakePaymentCardComponent implements OnChanges {
  @Input() paymentSucceeded = false;
  @Input() timezone: string = undefined;
  @Input() payment: UpcomingPayment;
  @Input() customerSearch: CustomerSearchResponse;
  @Input() loading = false;
  @Input() paymentMethod: PaymentAccount;
  @Input() otherAmount: number;
  @Input() autopayEnrollment: AutoPayEnrollmentResponse;

  @Output() selectAmountClick = new EventEmitter<UpcomingPayment>();
  @Output() selectPaymentClick = new EventEmitter<UpcomingPayment>();
  @Output() makePaymentClick = new EventEmitter<UpcomingPayment>();
  policyTypeIcon: string;

  constructor(
    private readonly routerService: RouterService,
    private readonly alertCtrl: AlertController,
    private readonly analyticsService: AnalyticsService
  ) {}

  ngOnChanges(changes: SimpleChanges) {
    if (changes.payment && changes.payment.currentValue) {
      this.getPolicyTypeIconPath(this.payment.policyType);
    }
  }

  hasRegId() {
    if (
      this.customerSearch &&
      this.customerSearch.registrations &&
      this.customerSearch.registrations.length > 0
    ) {
      if (this.customerSearch.registrations.filter((reg) => reg.registrationId).length > 0) {
        return true;
      }
    }
    return false;
  }

  isOverdue(): boolean {
    if (!this.payment) {
      return false;
    }

    return this.payment.isPastDue;
  }

  getPolicyTypeIconPath(policyType: PolicyType): void {
    switch (policyType) {
      case PolicyType.PUP:
        this.policyTypeIcon = '/assets/csaa-mobile/vectors/icn-PUP.svg';
        break;
      case PolicyType.Auto:
        this.policyTypeIcon = '/assets/csaa-mobile/vectors/icn-Auto.svg';
        break;
    }
  }

  getAutopayInfo() {
    return PaymentHelper.getAutopayEnrollmentLabel(this.autopayEnrollment);
  }

  async goToAutopaySettings() {
    if (!this.isOverdue()) {
      return this.routerService
        .navigateForward('csaa.payment.autopay.settings', {
          policyNumber: this.payment.policyNumber,
        })
        .then(noop);
    }

    this.analyticsService.trackEvent(EventName.AUTOPAY_ENROLLMENT_STOPPED, Category.payments, {
      event_type: EventType.OPTION_SELECTED,
      ...AnalyticsService.mapPaymentPolicy(this.payment),
    });
    const alert = await this.alertCtrl.create({
      header: 'Unable to Proceed',
      message:
        'We are unable to process your AutoPay enrollment request due to a past due balance on your account.' +
        ' To continue, you will need to make a payment on your account.',
      buttons: ['OK'],
    });
    return await alert.present();
  }
}
