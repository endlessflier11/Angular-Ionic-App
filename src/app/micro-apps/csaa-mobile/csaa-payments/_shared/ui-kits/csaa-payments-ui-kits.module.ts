import { NgModule } from '@angular/core';
import { IonicModule } from '@ionic/angular';
import { CommonModule } from '@angular/common';
import { CsaaPaymentHistoryCardComponent } from './csaa-payment-history-card/csaa-payment-history-card.component';
import { UiKitsModule } from '../../../_core/ui-kits/ui-kits.module';
import { CsaaPaymentHistorySelectCardComponent } from './csaa-payment-history-select-card/csaa-payment-history-select-card.component';
import { CsaaPaymentSucceededCardComponent } from './csaa-payment-succeeded-card/csaa-payment-succeeded-card.component';
import { CsaaMakePaymentCardComponent } from './csaa-make-payment-card/csaa-make-payment-card.component';
import { CsaaPayAllPoliciesCardComponent } from './csaa-pay-all-policies-card/csaa-pay-all-policies-card.component';
import { PaymentTermsConditionsComponent } from './payment-terms-conditions/payment-terms-conditions.component';
import { AutopayTermsConditionsComponent } from './autopay-terms-conditions/autopay-terms-conditions.component';
import { CsaaBillsCardComponent } from './csaa-bills-card/csaa-bills-card.component';
import { AutopayDiscountCardComponent } from './autopay-discount-card/autopay-discount-card.component';

const SHARED_COMPONENTS = [
  CsaaPaymentHistoryCardComponent,
  CsaaPaymentHistorySelectCardComponent,
  CsaaMakePaymentCardComponent,
  CsaaPaymentSucceededCardComponent,
  CsaaPayAllPoliciesCardComponent,
  PaymentTermsConditionsComponent,
  AutopayTermsConditionsComponent,
  CsaaBillsCardComponent,
  AutopayDiscountCardComponent,
];

@NgModule({
  declarations: [...SHARED_COMPONENTS],
  imports: [CommonModule, IonicModule, UiKitsModule, IonicModule],
  exports: [...SHARED_COMPONENTS],
})
export class CsaaPaymentsUiKitsModule {}
