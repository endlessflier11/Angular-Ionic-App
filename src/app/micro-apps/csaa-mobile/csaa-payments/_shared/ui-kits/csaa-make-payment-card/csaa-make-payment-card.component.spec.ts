import { CommonModule } from '@angular/common';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { IonicModule } from '@ionic/angular';
import { By } from '../../../../../../../testing';
import { EMPLOYEE_SEARCH } from '../../../../../../../testing/fixtures/make-payments-page/employee-search.fixture';
import { MAKE_PAYMENT_METHOD } from '../../../../../../../testing/fixtures/make-payments-page/make-payments-page.fixture';
import { PageTestingModule } from '../../../../../../../testing/misc/page-testing.module';
import { PaymentType } from '../../../../_core/interfaces/payment.interface';
import { PolicyType } from '../../../../_core/interfaces/policy.interface';
import DateHelper from '../../../../_core/shared/date.helper';
import { UiKitsModule } from '../../../../_core/ui-kits/ui-kits.module';
import { CsaaMakePaymentCardComponent } from './csaa-make-payment-card.component';

describe('CsaaMakePaymentCardComponent', () => {
  const DATE_NOW = Date.parse('2021-01-01');
  let component: CsaaMakePaymentCardComponent;
  let fixture: ComponentFixture<CsaaMakePaymentCardComponent>;

  beforeEach(async () => {
    jest.spyOn(Date, 'now').mockReturnValue(DATE_NOW);
    TestBed.configureTestingModule({
      declarations: [CsaaMakePaymentCardComponent],
      imports: [
        CommonModule,
        IonicModule,
        UiKitsModule,
        PageTestingModule.withConfig({
          providesStorage: true,
          providesAnalytics: true,
          providesConfig: true,
        }),
      ],
      schemas: [NO_ERRORS_SCHEMA],
    }).compileComponents();

    fixture = TestBed.createComponent(CsaaMakePaymentCardComponent);
    component = fixture.componentInstance;
    const tzOffset = new Date().getTimezoneOffset() / 60;
    component.timezone =
      (tzOffset > 0 ? '-' : '+') + tzOffset.toString().padStart(2, '0').padEnd(4, '0');

    component.paymentMethod = MAKE_PAYMENT_METHOD;
    component.customerSearch = EMPLOYEE_SEARCH;
    fixture.detectChanges();
  });

  afterEach(() => {
    jest.spyOn(Date, 'now').mockRestore();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshots when payment is due', () => {
    component.payment = {
      vehicles: ['Toyota Prius'],
      subtitle: 'test',
      autoPay: true,
      minimumAmount: 20,
      remainingPremium: 10000,
      otherAmount: 1000,
      type: PaymentType.other,
      nextDueDate: DateHelper.toDate('2040-08-19'),
      nextAmount: 30,
      allPolicies: false,
      policyNumber: 'AZSS294857564',
      policyType: PolicyType.Auto,
      dueDate: DateHelper.toDate('2040-08-19'),
      isPastDue: false,
      isPaymentDue: true,
      amount: 148.1,
      policyRiskState: 'test',
      autopayEnrollment: null,
    };
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
  it('should match snapshots when payment is overdue', () => {
    component.payment = {
      vehicles: ['Lexus NX', 'BMW X1'],
      subtitle: 'test',
      autoPay: true,
      minimumAmount: 20,
      remainingPremium: 10000,
      otherAmount: 1000,
      type: PaymentType.other,
      nextDueDate: DateHelper.toDate('2040-08-19'),
      nextAmount: 30,
      allPolicies: false,
      policyNumber: 'AZSS294857564',
      policyType: PolicyType.Auto,
      dueDate: DateHelper.toDate('2018-08-10'),
      isPastDue: true,
      isPaymentDue: true,
      amount: 260.75,
      policyRiskState: 'test',
      autopayEnrollment: null,
    };
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
  it('it should display selectAmountClick when clicking on card details', () => {
    component.payment = {
      vehicles: ['Toyota Prius'],
      subtitle: 'test',
      autoPay: true,
      minimumAmount: 20,
      remainingPremium: 10000,
      otherAmount: 1000,
      type: PaymentType.other,
      nextDueDate: DateHelper.toDate('2040-08-19'),
      nextAmount: 30,
      allPolicies: false,
      policyNumber: 'AZSS294857564',
      policyType: PolicyType.Auto,
      dueDate: DateHelper.toDate('2040-08-19'),
      isPastDue: false,
      isPaymentDue: true,
      amount: 148.1,
      policyRiskState: 'test',
      autopayEnrollment: null,
    };
    fixture.detectChanges();
    const button = fixture.debugElement.query(By.css('.csaa-payment-amount-list-item'));
    button.triggerEventHandler('click', null);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('it should display selectPaymentClick when clicking on card details', () => {
    component.payment = {
      vehicles: ['Toyota Prius'],
      subtitle: 'test',
      autoPay: false,
      minimumAmount: 20,
      remainingPremium: 10000,
      otherAmount: 1000,
      type: PaymentType.other,
      nextDueDate: DateHelper.toDate('2040-08-19'),
      nextAmount: 30,
      allPolicies: false,
      policyNumber: 'AZSS294857564',
      policyType: PolicyType.Auto,
      dueDate: DateHelper.toDate('2040-08-19'),
      isPastDue: false,
      isPaymentDue: true,
      amount: 148.1,
      policyRiskState: 'test',
      autopayEnrollment: null,
    };
    fixture.detectChanges();
    const button = fixture.debugElement.query(By.css('.csaa-payment-method-list-item'));
    button.triggerEventHandler('click', null);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('it should display autopay when clicking on card details', () => {
    component.payment = {
      vehicles: ['Toyota Prius'],
      subtitle: 'test',
      autoPay: true,
      minimumAmount: 20,
      remainingPremium: 10000,
      otherAmount: 1000,
      type: PaymentType.other,
      nextDueDate: DateHelper.toDate('2040-08-19'),
      nextAmount: 30,
      allPolicies: false,
      policyNumber: 'AZSS294857564',
      policyType: PolicyType.Auto,
      dueDate: DateHelper.toDate('2040-08-19'),
      isPastDue: false,
      isPaymentDue: true,
      amount: 148.1,
      policyRiskState: 'test',
      autopayEnrollment: null,
    };
    fixture.detectChanges();
    const autoPayButton = fixture.debugElement.query(By.css('.csaa-payment-autopay-item'));
    expect(autoPayButton).toBeTruthy();
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('it should click on the make a payment button', () => {
    component.payment = {
      vehicles: ['Toyota Prius'],
      subtitle: 'test',
      autoPay: true,
      minimumAmount: 20,
      remainingPremium: 10000,
      otherAmount: 1000,
      type: PaymentType.other,
      nextDueDate: DateHelper.toDate('2040-08-19'),
      nextAmount: 30,
      allPolicies: false,
      policyNumber: 'AZSS294857564',
      policyType: PolicyType.Auto,
      dueDate: DateHelper.toDate('2040-08-19'),
      isPastDue: false,
      isPaymentDue: true,
      amount: 148.1,
      policyRiskState: 'test',
      autopayEnrollment: null,
    };
    fixture.detectChanges();
    const makeAPaymentButton = fixture.debugElement.query(
      By.cssAndText('ion-button', 'Make a Payment')
    );
    expect(makeAPaymentButton).toBeTruthy();
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
