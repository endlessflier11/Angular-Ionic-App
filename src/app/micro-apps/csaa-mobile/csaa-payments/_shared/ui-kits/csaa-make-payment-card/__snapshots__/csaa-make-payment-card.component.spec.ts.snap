// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CsaaMakePaymentCardComponent it should click on the make a payment button 1`] = `
<csaa-make-payment-card
  __ngContext__={[Function LRootView]}
  alertCtrl={[Function AlertController]}
  analyticsService={[Function AnalyticsMockService]}
  customerSearch={[Function Object]}
  loading="false"
  makePaymentClick={[Function EventEmitter_]}
  payment={[Function Object]}
  paymentMethod={[Function Object]}
  paymentSucceeded="false"
  routerService={[Function RouterService]}
  selectAmountClick={[Function EventEmitter_]}
  selectPaymentClick={[Function EventEmitter_]}
  timezone={[Function String]}
>
  <ion-card
    class="csaa-card"
  >
    <ion-card-header>
      <ion-card-title>
        AZSS294857564
      </ion-card-title>
    </ion-card-header>
    <ion-card-content>
      <ion-grid>
        <ion-row
          class="item"
        >
          <ion-col>
            <div
              class="policy-meta-content"
            >
              <h3>
                Due:
                 Aug 19, 2040 
              </h3>
              <span
                class="subtitle"
              >
                <csaa-policy-subtitles>
                  <span
                    class="subtitle"
                  >
                    test
                  </span>
                </csaa-policy-subtitles>
              </span>
            </div>
          </ion-col>
          <ion-col
            size="auto"
          >
            <div
              class="policy-meta-icon"
            >
              <csaa-theme-icon
                slot="end"
              >
                <img
                  src="/assets/csaa-mobile/vectors/icn-Auto.svg"
                  style="width: 40px; height: 40px;"
                />
              </csaa-theme-icon>
            </div>
          </ion-col>
        </ion-row>
      </ion-grid>
      <ion-list>
        <ion-item
          class="csaa-payment-amount-list-item ion-no-padding"
          detail=""
        >
          <ion-label>
            Amount 
          </ion-label>
          <span
            slot="end"
          >
            $1,000.00
          </span>
        </ion-item>
        <ion-item
          class="csaa-payment-autopay-item ion-no-padding"
          detail=""
        >
          <ion-label>
            <label>
              AutoPay
            </label>
            <small>
            </small>
          </ion-label>
          <span
            slot="end"
          >
             On
          </span>
        </ion-item>
      </ion-list>
      <ion-button
        class="ion-no-margin"
        expand="full"
      >
         Make a Payment 
      </ion-button>
    </ion-card-content>
  </ion-card>
</csaa-make-payment-card>
`;

exports[`CsaaMakePaymentCardComponent it should display autopay when clicking on card details 1`] = `
<csaa-make-payment-card
  __ngContext__={[Function LRootView]}
  alertCtrl={[Function AlertController]}
  analyticsService={[Function AnalyticsMockService]}
  customerSearch={[Function Object]}
  loading="false"
  makePaymentClick={[Function EventEmitter_]}
  payment={[Function Object]}
  paymentMethod={[Function Object]}
  paymentSucceeded="false"
  routerService={[Function RouterService]}
  selectAmountClick={[Function EventEmitter_]}
  selectPaymentClick={[Function EventEmitter_]}
  timezone={[Function String]}
>
  <ion-card
    class="csaa-card"
  >
    <ion-card-header>
      <ion-card-title>
        AZSS294857564
      </ion-card-title>
    </ion-card-header>
    <ion-card-content>
      <ion-grid>
        <ion-row
          class="item"
        >
          <ion-col>
            <div
              class="policy-meta-content"
            >
              <h3>
                Due:
                 Aug 19, 2040 
              </h3>
              <span
                class="subtitle"
              >
                <csaa-policy-subtitles>
                  <span
                    class="subtitle"
                  >
                    test
                  </span>
                </csaa-policy-subtitles>
              </span>
            </div>
          </ion-col>
          <ion-col
            size="auto"
          >
            <div
              class="policy-meta-icon"
            >
              <csaa-theme-icon
                slot="end"
              >
                <img
                  src="/assets/csaa-mobile/vectors/icn-Auto.svg"
                  style="width: 40px; height: 40px;"
                />
              </csaa-theme-icon>
            </div>
          </ion-col>
        </ion-row>
      </ion-grid>
      <ion-list>
        <ion-item
          class="csaa-payment-amount-list-item ion-no-padding"
          detail=""
        >
          <ion-label>
            Amount 
          </ion-label>
          <span
            slot="end"
          >
            $1,000.00
          </span>
        </ion-item>
        <ion-item
          class="csaa-payment-autopay-item ion-no-padding"
          detail=""
        >
          <ion-label>
            <label>
              AutoPay
            </label>
            <small>
            </small>
          </ion-label>
          <span
            slot="end"
          >
             On
          </span>
        </ion-item>
      </ion-list>
      <ion-button
        class="ion-no-margin"
        expand="full"
      >
         Make a Payment 
      </ion-button>
    </ion-card-content>
  </ion-card>
</csaa-make-payment-card>
`;

exports[`CsaaMakePaymentCardComponent it should display selectAmountClick when clicking on card details 1`] = `
<csaa-make-payment-card
  __ngContext__={[Function LRootView]}
  alertCtrl={[Function AlertController]}
  analyticsService={[Function AnalyticsMockService]}
  customerSearch={[Function Object]}
  loading="false"
  makePaymentClick={[Function EventEmitter_]}
  payment={[Function Object]}
  paymentMethod={[Function Object]}
  paymentSucceeded="false"
  routerService={[Function RouterService]}
  selectAmountClick={[Function EventEmitter_]}
  selectPaymentClick={[Function EventEmitter_]}
  timezone={[Function String]}
>
  <ion-card
    class="csaa-card"
  >
    <ion-card-header>
      <ion-card-title>
        AZSS294857564
      </ion-card-title>
    </ion-card-header>
    <ion-card-content>
      <ion-grid>
        <ion-row
          class="item"
        >
          <ion-col>
            <div
              class="policy-meta-content"
            >
              <h3>
                Due:
                 Aug 19, 2040 
              </h3>
              <span
                class="subtitle"
              >
                <csaa-policy-subtitles>
                  <span
                    class="subtitle"
                  >
                    test
                  </span>
                </csaa-policy-subtitles>
              </span>
            </div>
          </ion-col>
          <ion-col
            size="auto"
          >
            <div
              class="policy-meta-icon"
            >
              <csaa-theme-icon
                slot="end"
              >
                <img
                  src="/assets/csaa-mobile/vectors/icn-Auto.svg"
                  style="width: 40px; height: 40px;"
                />
              </csaa-theme-icon>
            </div>
          </ion-col>
        </ion-row>
      </ion-grid>
      <ion-list>
        <ion-item
          class="csaa-payment-amount-list-item ion-no-padding"
          detail=""
        >
          <ion-label>
            Amount 
          </ion-label>
          <span
            slot="end"
          >
            $1,000.00
          </span>
        </ion-item>
        <ion-item
          class="csaa-payment-autopay-item ion-no-padding"
          detail=""
        >
          <ion-label>
            <label>
              AutoPay
            </label>
            <small>
            </small>
          </ion-label>
          <span
            slot="end"
          >
             On
          </span>
        </ion-item>
      </ion-list>
      <ion-button
        class="ion-no-margin"
        expand="full"
      >
         Make a Payment 
      </ion-button>
    </ion-card-content>
  </ion-card>
</csaa-make-payment-card>
`;

exports[`CsaaMakePaymentCardComponent it should display selectPaymentClick when clicking on card details 1`] = `
<csaa-make-payment-card
  __ngContext__={[Function LRootView]}
  alertCtrl={[Function AlertController]}
  analyticsService={[Function AnalyticsMockService]}
  customerSearch={[Function Object]}
  loading="false"
  makePaymentClick={[Function EventEmitter_]}
  payment={[Function Object]}
  paymentMethod={[Function Object]}
  paymentSucceeded="false"
  routerService={[Function RouterService]}
  selectAmountClick={[Function EventEmitter_]}
  selectPaymentClick={[Function EventEmitter_]}
  timezone={[Function String]}
>
  <ion-card
    class="csaa-card"
  >
    <ion-card-header>
      <ion-card-title>
        AZSS294857564
      </ion-card-title>
    </ion-card-header>
    <ion-card-content>
      <ion-grid>
        <ion-row
          class="item"
        >
          <ion-col>
            <div
              class="policy-meta-content"
            >
              <h3>
                Due:
                 Aug 19, 2040 
              </h3>
              <span
                class="subtitle"
              >
                <csaa-policy-subtitles>
                  <span
                    class="subtitle"
                  >
                    test
                  </span>
                </csaa-policy-subtitles>
              </span>
            </div>
          </ion-col>
          <ion-col
            size="auto"
          >
            <div
              class="policy-meta-icon"
            >
              <csaa-theme-icon
                slot="end"
              >
                <img
                  src="/assets/csaa-mobile/vectors/icn-Auto.svg"
                  style="width: 40px; height: 40px;"
                />
              </csaa-theme-icon>
            </div>
          </ion-col>
        </ion-row>
      </ion-grid>
      <ion-list>
        <ion-item
          class="csaa-payment-amount-list-item ion-no-padding"
          detail=""
        >
          <ion-label>
            Amount 
          </ion-label>
          <span
            slot="end"
          >
            $1,000.00
          </span>
        </ion-item>
        <ion-item
          class="csaa-payment-method-list-item ion-no-padding"
          detail=""
          lines="none"
        >
          <ion-avatar
            slot="start"
          >
            <span>
              <img
                src="/assets/csaa-mobile/vectors/mastercard.svg"
              />
            </span>
          </ion-avatar>
          <ion-label
            class="ion-text-nowrap"
          >
            <span>
              Matts test 5454 5454 5454
            </span>
          </ion-label>
          <span
            slot="end"
          >
            Debit
          </span>
        </ion-item>
        <ion-item
          class="csaa-payment-autopay-item ion-no-padding"
          detail=""
        >
          <ion-label>
            <label>
              AutoPay
            </label>
          </ion-label>
          <span
            slot="end"
          >
             Off
          </span>
        </ion-item>
      </ion-list>
      <ion-button
        class="ion-no-margin"
        expand="full"
      >
         Make a Payment 
      </ion-button>
    </ion-card-content>
  </ion-card>
</csaa-make-payment-card>
`;

exports[`CsaaMakePaymentCardComponent should match snapshots when payment is due 1`] = `
<csaa-make-payment-card
  __ngContext__={[Function LRootView]}
  alertCtrl={[Function AlertController]}
  analyticsService={[Function AnalyticsMockService]}
  customerSearch={[Function Object]}
  loading="false"
  makePaymentClick={[Function EventEmitter_]}
  payment={[Function Object]}
  paymentMethod={[Function Object]}
  paymentSucceeded="false"
  routerService={[Function RouterService]}
  selectAmountClick={[Function EventEmitter_]}
  selectPaymentClick={[Function EventEmitter_]}
  timezone={[Function String]}
>
  <ion-card
    class="csaa-card"
  >
    <ion-card-header>
      <ion-card-title>
        AZSS294857564
      </ion-card-title>
    </ion-card-header>
    <ion-card-content>
      <ion-grid>
        <ion-row
          class="item"
        >
          <ion-col>
            <div
              class="policy-meta-content"
            >
              <h3>
                Due:
                 Aug 19, 2040 
              </h3>
              <span
                class="subtitle"
              >
                <csaa-policy-subtitles>
                  <span
                    class="subtitle"
                  >
                    test
                  </span>
                </csaa-policy-subtitles>
              </span>
            </div>
          </ion-col>
          <ion-col
            size="auto"
          >
            <div
              class="policy-meta-icon"
            >
              <csaa-theme-icon
                slot="end"
              >
                <img
                  src="/assets/csaa-mobile/vectors/icn-Auto.svg"
                  style="width: 40px; height: 40px;"
                />
              </csaa-theme-icon>
            </div>
          </ion-col>
        </ion-row>
      </ion-grid>
      <ion-list>
        <ion-item
          class="csaa-payment-amount-list-item ion-no-padding"
          detail=""
        >
          <ion-label>
            Amount 
          </ion-label>
          <span
            slot="end"
          >
            $1,000.00
          </span>
        </ion-item>
        <ion-item
          class="csaa-payment-autopay-item ion-no-padding"
          detail=""
        >
          <ion-label>
            <label>
              AutoPay
            </label>
            <small>
            </small>
          </ion-label>
          <span
            slot="end"
          >
             On
          </span>
        </ion-item>
      </ion-list>
      <ion-button
        class="ion-no-margin"
        expand="full"
      >
         Make a Payment 
      </ion-button>
    </ion-card-content>
  </ion-card>
</csaa-make-payment-card>
`;

exports[`CsaaMakePaymentCardComponent should match snapshots when payment is overdue 1`] = `
<csaa-make-payment-card
  __ngContext__={[Function LRootView]}
  alertCtrl={[Function AlertController]}
  analyticsService={[Function AnalyticsMockService]}
  customerSearch={[Function Object]}
  loading="false"
  makePaymentClick={[Function EventEmitter_]}
  payment={[Function Object]}
  paymentMethod={[Function Object]}
  paymentSucceeded="false"
  routerService={[Function RouterService]}
  selectAmountClick={[Function EventEmitter_]}
  selectPaymentClick={[Function EventEmitter_]}
  timezone={[Function String]}
>
  <ion-card
    class="csaa-card overdue"
  >
    <ion-card-header>
      <ion-card-title>
        AZSS294857564
      </ion-card-title>
    </ion-card-header>
    <ion-card-content>
      <ion-grid>
        <ion-row
          class="item"
        >
          <ion-col>
            <div
              class="policy-meta-content"
            >
              <h3>
                Past Due
                 Aug 10, 2018 
              </h3>
              <span
                class="subtitle"
              >
                <csaa-policy-subtitles>
                  <span
                    class="subtitle"
                  >
                    test
                  </span>
                </csaa-policy-subtitles>
              </span>
            </div>
          </ion-col>
          <ion-col
            size="auto"
          >
            <div
              class="policy-meta-icon"
            >
              <csaa-theme-icon
                slot="end"
              >
                <img
                  src="/assets/csaa-mobile/vectors/icn-Auto.svg"
                  style="width: 40px; height: 40px;"
                />
              </csaa-theme-icon>
            </div>
          </ion-col>
        </ion-row>
      </ion-grid>
      <ion-list>
        <ion-item
          class="csaa-payment-amount-list-item ion-no-padding"
          detail=""
        >
          <ion-label>
            Amount 
          </ion-label>
          <span
            slot="end"
          >
            $1,000.00
          </span>
        </ion-item>
        <ion-item
          class="csaa-payment-autopay-item ion-no-padding"
          detail=""
        >
          <ion-label>
            <label>
              AutoPay
            </label>
            <small>
            </small>
          </ion-label>
          <span
            slot="end"
          >
             On
          </span>
        </ion-item>
      </ion-list>
      <ion-button
        class="ion-no-margin"
        expand="full"
      >
         Make a Payment 
      </ion-button>
    </ion-card-content>
  </ion-card>
</csaa-make-payment-card>
`;
